parser_library/test/workspace_manager_test.cpp

```diff
@@ -15,87 +15,107 @@
 #include "gtest/gtest.h"
 
 #include "common_testing.h"
+#include "debugging/debugger_configuration.h"
 #include "lib_config.h"
 #include "message_consumer_mock.h"
 #include "nlohmann/json.hpp"
 #include "utils/path.h"
 #include "utils/platform.h"
 #include "workspace/consume_diagnostics_mock.h"
 #include "workspace_manager.h"
+#include "workspace_manager_response.h"
+#include "workspace_manager_response_mock.h"
 
 using namespace hlasm_plugin::parser_library;
 
 TEST(workspace_manager, add_not_existing_workspace)
 {
-    workspace_manager ws_mngr;
+    auto ws_mngr = create_workspace_manager();
     diag_consumer_mock consumer;
-    ws_mngr.register_diagnostics_consumer(&consumer);
+    ws_mngr->register_diagnostics_consumer(&consumer);
 
-    ws_mngr.add_workspace("workspace", "not_existing");
+    ws_mngr->add_workspace("workspace", "not_existing");
 
-    size_t count = ws_mngr.get_workspaces_count();
-    EXPECT_EQ(count, (size_t)1);
+    ws_mngr->idle_handler();
 
     EXPECT_EQ(consumer.diags.diagnostics_size(), (size_t)0);
 }
 
 TEST(workspace_manager, add_existing_workspace)
 {
-    workspace_manager ws_mngr;
+    auto ws_mngr = create_workspace_manager();
     diag_consumer_mock consumer;
-    ws_mngr.register_diagnostics_consumer(&consumer);
+    ws_mngr->register_diagnostics_consumer(&consumer);
 
-    ws_mngr.add_workspace("workspace", "test/library/test_wks");
+    ws_mngr->add_workspace("workspace", "test/library/test_wks");
 
-    size_t count = ws_mngr.get_workspaces_count();
-    EXPECT_EQ(count, (size_t)1);
+    ws_mngr->idle_handler();
 
     EXPECT_EQ(consumer.diags.diagnostics_size(), (size_t)0);
 }
```

Comment:
Is there any difference between these two tests? The different URIs don't really matter, or do they somehow?

Reply:
Oh, I see. Thx!